cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ScheduleSpecificationCodec.java

```diff
@@ -32,11 +34,48 @@
  *
  */
 public class ScheduleSpecificationCodec extends AbstractSpecificationCodec<ScheduleSpecification> {
+
+  /**
+   * Schedule Type.
+   */
+  private enum ScheduleType {
+    ORIGINAL_TIME,
```

Comment:
`TimeSchedule` and `Schedule` both are cron based schedules. So seems like duplication to me to have ScheduleType `ORIGINAL_TIME` and `TIME` since they both serve the purpose of identifying schedules based on the crontab. Can we handle the conversion of `Schedule` to `TimeSchedule` during serialization/deserialization process?